tests/modules/basics.c

```diff
@@ -407,6 +407,26 @@ int TestStringAppendAM(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
     return REDISMODULE_OK;
 }
 
+/* TEST.STRING.TRIM -- Test we trim a string with free space. */
+int TestTrimString(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    REDISMODULE_NOT_USED(argv);
+    REDISMODULE_NOT_USED(argc);
+    RedisModuleString *s = RedisModule_CreateString(ctx,"foo",3);
+    char *tmp = RedisModule_Alloc(1024);
+    RedisModule_StringAppendBuffer(ctx,s,tmp,1024);
```

Comment:
```
48834:M 01 Mar 2023 00:30:32.025 # <test> Testing test.string.trim
48834:M 01 Mar 2023 00:30:32.025 # <test> Test error reply: String was not trimmed as expected.
```
https://github.com/redis/redis/actions/runs/4298577682/jobs/7492838351#step:6:249

This will fail on MacOS, and it will also fail locally. 
Looks like realloc doesn't shrink memory in this cas.
I tested that setting the number to 1300 can pass the test. 
Any other better idea? or reasons?

Reply:
we probably need to skip that test when the allocator is not jemalloc

Reply:
let me change it in #11878?